gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/common/AbstractServiceCallableFactoryClassComposer.java

```diff
@@ -60,8 +61,17 @@ protected TransportContext getTransportContext() {
     return transportContext;
   }
 
+  protected Transport getTransport() {
+    return Transport.GRPC;
+  }
+
   @Override
   public GapicClass generate(GapicContext context, Service service) {
+    // Do not generate the Callable Factory if there are no RPCs enabled for the Transport
+    if (!service.hasAnyEnabledMethodsForTransport(getTransport())) {
+      return GapicClass.createEmpty();
```

Comment:
How bad is it gonna be if we return `null` here instead of creating an empty `GapicClass`? Or if we move the check to before calling this method? I would prefer to not returning anything or check before calling this method, because an `GapicClass.createEmpty()` could mean an empty file rather than no files. 
Another option is to change the return type to `Optional<GapicClass>` so that we can return `Optional.empty()` here but that apparently is a much larger effort.

Reply:
I don't think it's bad -- I just wanted to avoid passing around nulls if possible. Maybe this could be a possible use case for null?

> GapicClass.createEmpty() could mean an empty file rather than no files

I wonder if it would make sense if I created something like `GapicClass.shouldGenerateClass()` that is generated as `true` except for this case (i.e. in a `GapicClass.createNonGenerateableClass()` func or something).

> Another option is to change the return type to Optional<GapicClass> so that we can return Optional.empty() here but that apparently is a much larger effort.

I think that also makes sense, but yeah would require a larger change.